src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +89,55 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
-            // Description
+            // DisplayFormatString
+            if (displayFormatAttribute != null)
+            {
+                displayMetadata.DisplayFormatString = displayFormatAttribute.DataFormatString;
+            }
+
+            // DisplayDescription
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
+                    !string.IsNullOrEmpty(displayAttribute.Name) &&
+                    displayAttribute.ResourceType == null)
```

Comment:
What I meant is to cover the case where both `localizer` and `displayAttribute.ResourceType` are non-`null`. Those tests cover the case where `ResourceType` is non-`null` but `localizer` there is `null`.